Floofbot/Services/RaidProtectionService.cs

```diff
@@ -0,0 +1,387 @@
+            // we run an async task to remove their point after the specified duration
+            UserMessageCountTimeout(msg.Author.Id);
+            return false;
+        }
+        public async Task<bool> CheckMentions(SocketUserMessage msg, IGuild guild, SocketRole modRole, ITextChannel modChannel)
+        {
+            if (msg.MentionedUsers.Count > 10)
+            {
+                string reason = "Raid Protection =>" + msg.MentionedUsers.Count + " mentions in one message.";
+                try
+                {
+                    //sends message to user
+                    EmbedBuilder builder = new EmbedBuilder();
+                    builder.Title = "⚖️ Ban Notification";
+                    builder.Description = $"You have been banned from {guild.Name}";
+                    builder.AddField("Reason", reason);
+                    builder.Color = Discord.Color.Red;
+                    await msg.Author.SendMessageAsync("", false, builder.Build());
+                    await guild.AddBanAsync(msg.Author, 0, reason);
+
+                    await NotifyModerators(modRole, modChannel, " I have banned a user for mentioning " + msg.MentionedUsers.Count() + " members");
+                }
+                catch (Exception e)
+                {
+                    Log.Error("Error banning user for mass mention: " + e);
+                }
+                return true;
+            }
+            return false;
+        }
+        public async Task<bool> CheckLetterSpam(SocketMessage msg)
+        {
+            var matches = Regex.Matches(msg.Content, @"(.)\1+");
+            foreach (Match m in matches)
+            {
+                // string has more than 5 letters in a row
+                if (m.Length > 5)
+                {
+                    // add a bad boye point for the user
+                    if (userPunishmentCount.ContainsKey(msg.Author.Id))
+                    {
+                        userPunishmentCount[msg.Author.Id] += 1;
+                    }
+                    else // they were a good boye but now they are not
+                    {
+                        userPunishmentCount.Add(msg.Author.Id, 1);
+                    }
+                    // we run an async task to remove their point after the specified duration
+                    UserPunishmentTimeout(msg.Author.Id);
+                    var botMsg = await msg.Channel.SendMessageAsync(msg.Author.Mention + " no spamming!");
+                    await Task.Delay(botMessageDeletionDelay);
+                    await botMsg.DeleteAsync();
+                    // we return here because we only need to check for at least one match, doesnt matter if there are more
+                    return true;
+                }                     
+            }
+            return false;
+
+        }
+        public async Task<bool> CheckInviteLinks(SocketMessage msg)
+        {
+            if (msg.Content.Contains("discord.gg") || msg.Content.Contains("d.gg"))
```

Comment:
This is a little too wide of a net (I'd also like to point out that the first half of this statement is redundant)

I'll have to come back and come up with a better way to parse out invite links, but be aware that this would catch an innocent https://discord.gg link and miss https://discord.com/invite/rfurry

Reply:
Added a regex algorithm for this